lib/rubygems/commands/query_command.rb

```diff
@@ -129,55 +137,55 @@ def display_header(type)
 
   #Guts of original execute
   def show_gems(name, prerelease)
-    req = Gem::Requirement.default
-    # TODO: deprecate for real
-    dep = Gem::Deprecate.skip_during { Gem::Dependency.new name, req }
-    dep.prerelease = prerelease
```

Comment:
I'm unsure about this deprecation but the current command is broken for local gems and the `--no-prerelease` option.

If we keep the above three lines and add `specs.select! { |spec| dep.match?(spec.name, spec.version) }` after the initial filtering of local specs, namely, where we do:

```ruby
specs = Gem::Specification.find_all do |s|
  s.name =~ name and req =~ s.version
end
```

Then I think that would fix that bug and we can remove the warning about it

```ruby
if prerelease? && !both?
  alert_warning("prereleases are always shown locally")
end
```

Reply:
@deivid-rodriguez how can i reproduce the bug?

Reply:
```
$ ruby -Ilib bin/gem list bund

*** LOCAL GEMS ***

bundler (2.1.0.pre.1, 2.0.1, default: 1.17.3)

$ ruby -Ilib bin/gem query bund --prerelease
WARNING:  prereleases are always shown locally

*** LOCAL GEMS ***

bundler (2.1.0.pre.1, 2.0.1, default: 1.17.3)

$ ruby -Ilib bin/gem query bund --no-prerelease

*** LOCAL GEMS ***

bundler (2.1.0.pre.1, 2.0.1, default: 1.17.3)
```

If I pass the `--no-prerelease` parameter, prereleases should be excluded.